beacon-chain/sync/backfill/verify_column.go

```diff
@@ -0,0 +1,189 @@
+package backfill
+
+import (
+	"io"
+
+	"github.com/OffchainLabs/prysm/v7/beacon-chain/core/peerdas"
+	"github.com/OffchainLabs/prysm/v7/beacon-chain/das"
+	"github.com/OffchainLabs/prysm/v7/consensus-types/blocks"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/pkg/errors"
+)
+
+type columnBisector struct {
+	rootKeys     map[[32]byte]rootKey
+	pidKeys      map[peer.ID]pidKey
+	columnSource map[rootKey]map[uint64]pidKey
+	bisected     map[pidKey][]blocks.RODataColumn
+	pidIter      []peer.ID
+	current      int
+	next         int
+	downscore    peerDownscorer
+	errs         []error
+	failures     map[rootKey]peerdas.ColumnIndices
+}
+
+type pidKey *peer.ID
+type rootKey *[32]byte
+
+var errColumnVerification = errors.New("column verification failed")
```

Comment:
Nit:

`var`s definition could be grouped. (As well as type for `pidKey` and `rootKey`.)